compiler/grainformat/comment_utils.re

```diff
@@ -0,0 +1,519 @@
+open Grain_diagnostics;
+module Doc = Res_doc;
+
+let rec get_comments_on_line = (line: int, comments) =>
+  if (List.length(comments) == 0) {
+    [];
+  } else {
+    let c: Grain_parsing.Parsetree.comment = List.hd(comments);
+    let c_loc: Grain_parsing.Location.t = Locations.get_comment_loc(c);
+    let (_, cmtline, cmtchar, _) =
+      Locations.get_raw_pos_info(c_loc.loc_start);
+
+    if (cmtline == line) {
+      [c] @ get_comments_on_line(line, List.tl(comments));
+    } else {
+      []; // can stop early as there will be no more
+    };
+  };
+
+let rec get_comments_before_line = (line: int, comments) =>
+  if (List.length(comments) == 0) {
+    [];
+  } else {
+    let c: Grain_parsing.Parsetree.comment = List.hd(comments);
+    let c_loc: Grain_parsing.Location.t = Locations.get_comment_loc(c);
+    let (_, cmtline, cmtchar, _) = Locations.get_raw_pos_info(c_loc.loc_end);
+    if (cmtline < line) {
+      [c] @ get_comments_before_line(line, List.tl(comments));
+    } else {
+      []; // can stop early as there will be no more
+    };
+  };
+
+let rec get_comments_after_line = (line: int, comments) =>
+  if (List.length(comments) == 0) {
+    [];
+  } else {
+    let c: Grain_parsing.Parsetree.comment = List.hd(comments);
+    let c_loc: Grain_parsing.Location.t = Locations.get_comment_loc(c);
+    let (_, cmtline, cmtchar, _) = Locations.get_raw_pos_info(c_loc.loc_end);
+    if (cmtline > line) {
+      [c] @ get_comments_after_line(line, List.tl(comments));
+    } else {
+      get_comments_after_line(
+        line,
+        List.tl(comments) // gotta keep checking
+      );
+    };
+  };
+
+let print_comments = (comments: list(Grain_parsing.Parsetree.comment)) => {
+  List.map(c => Comments.print_comment(c), comments);
+};
+
+let rec get_comments_between_lines = (line1: int, line2: int, comments) =>
+  if (List.length(comments) == 0) {
+    [];
+  } else {
+    let c: Grain_parsing.Parsetree.comment = List.hd(comments);
+    let c_loc: Grain_parsing.Location.t = Locations.get_comment_loc(c);
+    let (_, cmtsline, cmtschar, _) =
+      Locations.get_raw_pos_info(c_loc.loc_start);
+
+    let (_, cmteline, cmtechar, _) =
+      Locations.get_raw_pos_info(c_loc.loc_end);
+
+    let cmts =
+      if (cmtsline > line1) {
+        if (cmteline < line2) {
+          [c] @ get_comments_between_lines(line1, line2, List.tl(comments));
+        } else
+          {
+            [];
+          }; // can stop early
+      } else {
+        // it's before, so keep going
+        get_comments_between_lines(
+          line1,
+          line2,
+          List.tl(comments),
+        );
+      };
+    cmts;
+  };
+
+let rec get_comments_inside_location =
+        (
+          ~location: Grain_parsing.Location.t,
+          comments: list(Grain_parsing.Parsetree.comment),
+        ) => {
+  let (_, stmtStartLine, stmsStartChar, _) =
```

Comment:
The camel case strikes again!

Reply:
That damn camel!   I've gone through and removed them all I think